src/L2_projection.jl

```diff
@@ -160,68 +153,75 @@ The order of the returned data correspond to the order of the `L2Projector`s int
 `DofHandler`. To export the result, use `vtk_point_data(vtk, proj, projected_data)`.
 """
 function project(proj::L2Projector,
-                 vars::AbstractVector{<:AbstractVector{T}},
-                 qr_rhs::QuadratureRule) where T <: Union{Number, AbstractTensor}
-
-    # For using the deprecated API
-    fe_values = CellValues(qr_rhs, proj.func_ip, proj.geom_ip)
+                 vars::Union{AbstractVector{TC}, AbstractDict{Int, TC}},
+                 qrs_rhs::Vector{<:QuadratureRule}
+                 ) where {TC <: AbstractVector{T}} where T <: Union{Number, AbstractTensor}
 
-    M = T <: AbstractTensor ? length(vars[1][1].data) : 1
+    isclosed(proj) || error("The L2Projector is not closed")
+    length(qrs_rhs) == length(proj.dh.subdofhandlers) || error("Number of qrs_rhs must match the number of `add!`ed sets")
 
-    projected_vals = _project(vars, proj, fe_values, M, T)::Vector{T}
+    M = T <: AbstractTensor ? Tensors.n_components(Tensors.get_base(T)) : 1
 
-    return projected_vals
+    return _project(proj, qrs_rhs, vars, M, T)::Vector{T}
 end
-function project(p::L2Projector, vars::AbstractMatrix, qr_rhs::QuadratureRule)
+function project(p::L2Projector, vars::Union{AbstractVector, AbstractDict}, qr_rhs::QuadratureRule)
+    return project(p, vars, [qr_rhs])
+end
+function project(p::L2Projector, vars::AbstractMatrix, qr_rhs)
     # TODO: Random access into vars is required for now, hence the collect
     return project(p, collect(eachcol(vars)), qr_rhs)
 end
 
-function _project(vars, proj::L2Projector, fe_values::AbstractValues, M::Integer, ::Type{T}) where {T}
+function _project(proj::L2Projector, qrs_rhs::Vector{<:QuadratureRule}, vars::Union{AbstractVector, AbstractDict}, M::Integer, ::Type{T}) where T
+    f = zeros(ndofs(proj.dh), M)
+    for (sdh, qr_rhs) in zip(proj.dh.subdofhandlers, qrs_rhs)
```

Comment:
Maybe add a checks here that checks that length(qrs_rhs) as number of length(subdofhandle)r (and the same `refshape`)? CellValues will probable throw an error anyways, but could be nice to catch it earlier.